### PR TITLE
feat(test): Add more /force_auth functional tests.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Add for:
* Fx Desktop Sync v1
* OAuth
* OAuth web channel

Adding these tests to ensure force auth functionality does not regress with
Fennec related changes to how the brokers handle force auth.

issue #1933
issue #1934

### CODE REVIEW

File: tests/functional/lib/helpers.js [hunk 1]
```diff
@@ -329,24 +329,47 @@ define([
     return openFxaFromRp(context, page, urlSuffix, true);
   }
 
+  function reOpenWithAdditionalQueryParams(context, additionalQueryParams, waitForSelector) {
+    return context.remote
+      .getCurrentUrl()
+      .then(function (url) {
+        var parsedUrl = Url.parse(url);
+        var rpQueryParams = Querystring.parse(parsedUrl.search);
```
Comment (shane-tomlinson) [inline tests/functional/lib/helpers.js]: rename `rpQueryParams` to `currentQueryParams`

File: tests/functional/lib/helpers.js [hunk 2]
```diff
@@ -358,18 +381,9 @@ define([
 
       .then(function () {
         if (urlSuffix) {
```
Comment (shane-tomlinson) [inline tests/functional/lib/helpers.js]: `urlSuffix`=>`additionalQueryParams` for logical consistency.

File: tests/functional/lib/webchannel-helpers.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
```
Comment (shane-tomlinson) [inline tests/functional/lib/webchannel-helpers.js]: Add a header about this module's purpose.

**Comment by shane-tomlinson** [comment]
Ref #2771, #3094

**Comment by vladikoff** [comment]
:+1: woohoo more tests! tested locally and against latest

**Pull request merged**